File: atcoder/abc164C_gacha.cpp
// https://atcoder.jp/contests/abc164/tasks/abc164_c
#include<iostream>
#include<set>
#include<string>

using namespace std;

int main() {
    int n;
    cin >> n;

    set<string> values;
    string value;
    for (int i = 0; i < n; i++) {
        cin >> value;
        values.insert(value);
    }

    cout << values.size() << endl;
    return 0;
}

File: atcoder/abc164C_gacha.rs
// vicfred
// https://atcoder.jp/contests/abc164/tasks/abc164_c
// set
use std::io;
use std::collections::HashSet;

fn main() {
    let mut n = String::new();

    io::stdin()
        .read_line(&mut n)
        .unwrap();

    let n: i64 = n.trim().parse().unwrap();

    let mut kinds = HashSet::new();
    for _ in 0..n {
        let mut s = String::new();

        io::stdin()
            .read_line(&mut s)
            .unwrap();

        kinds.insert(s);
    }
    
    println!("{}", kinds.len());
}

